src/views/Pipeline.vue

```diff
@@ -367,7 +367,8 @@ export default defineComponent({
     },
     updateSelectedPinnedJob(jobEnumId: any) {
       const index = (this as any).selectedPinnedJobs.indexOf(jobEnumId);
-      if (index != -1) {
+      const pinnedJobs = new Set(this.getPinnedJobs);
+      if (index != -1 || !pinnedJobs.has(jobEnumId)) {
```

Comment:
It will have wrong result in case pinnedJobs is unpinned without using in filter